tests/python/pants_test/help/test_help_info_extracter.py

```diff
@@ -30,13 +32,15 @@ def do_test(args, kwargs, expected_display_args, expected_scoped_cmd_line_args):
     do_test(['-f', '--foo'], {'type': bool }, ['-f', '--[no-]foo'],
             ['-f', '--foo', '--no-foo'])
 
-    do_test(['--foo'], {}, ['--foo=<str>'], ['--foo'])
+    str_repr = 'newstr' if PY2 else 'str'
+
+    do_test(['--foo'], {}, ['--foo=<{}>'.format(str_repr)], ['--foo'])
```

Comment:
This fix gets the tests to pass, but I want to make sure this doesn't break a contract anyone is depending upon. 

The backport of `str` means in Py2 the help parameter's type will be `newstr`. Beyond the difference between unicode vs bytes, does anything depend on the representation being `str`? That is, is it fine that `newstr` is only str-like (duck typing). I would imagine this is fine and think the test has only hardcoded checking for `str` as a way to ensure the parameter is a string and not something like an int.

Reply:
AFAICT, this will actually end up changing `./pants help` to render `newstr`, which isn't ideal. Adding a special case for the rendering of this help would be preferable... maybe here: https://github.com/pantsbuild/pants/blob/039051735542d29ae02f4faa09c0c51c47292bf0/src/python/pants/help/help_info_extracter.py#L98

Reply:
You're right! Good catch, thanks.